src/index.js

```diff
@@ -40,12 +40,24 @@ const normalizeId = (id) => {
   return id;
 };
 
+const getEntries = ({ entries }) => {
```

Comment:
> First time I've seen that syntax, but it does confuse the heck out of me in terms of clean code.

I can change it, but it's pretty standard modern JS - IMHO nothing too fancy. 

> What happens when someone passes for example an array as options? Does it break with a nullpointer here?

It just desugars to:
```js
var getEntries = function getEntries(_ref) {
  var entries = _ref.entries;
  // ...
};
```
You can play around with it on [here](https://babeljs.io/repl#?babili=false&browsers=&build=&builtIns=false&spec=false&loose=false&code_lz=MYewdgzgLgBA5gUygUTFATgSwRGBeGACgG8YE0scYBfASnwD4Zjqg&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=react%2Cenv&prettier=false&targets=&version=7.6.4&externalPlugins=).

So in your described case, it will simply be `undefined` - because we can access arbitrary properties on arrays.

If anything - this shouldn't be guarded here, but when validating input - right at the start of the plugin factory. With JS it's not really ergonomic to validate everything against expected schema - so personally I would just leave it as is.

Reply:
That syntax if pretty standard use for ES6 these days.